binary_utils/src/lib.rs

```diff
@@ -40,6 +41,59 @@ impl FormatTime for LocalTimer {
     }
 }
 
+struct RotatingWriter {
+    path: PathBuf,
+    max_size: u64,
```

Comment:
You could use [`bytesize`](https://crates.io/crates/bytesize) for `max_size` type to make it more readable as it is done for [database](https://github.com/grandinetech/grandine/blob/develop/database/src/lib.rs#L102) module